src/envoy/mixer/config.cc

```diff
@@ -25,6 +25,9 @@ using ::istio::mixer_client::AttributesBuilder;
 using ::istio::mixer::v1::config::client::ServiceConfig;
 using ::istio::mixer::v1::config::client::TransportConfig;
 
+#define PROTOBUF_GET_STRING_OR_DEFAULT(message, field_name, default_value)                        \
+  ((message).has_##field_name() ? (message).field_name() : (default_value))
```

Comment:
protobuf primary type, such as string,  does not generate "has_"  function.  Only the message type does.
it has default value,  string default is empty.